stdlib/int16.mli

```diff
@@ -0,0 +1,171 @@
+(** Same as {!div}, except that arguments and result are interpreted as {e
+    unsigned} 16-bit integers. *)
+
+val rem : t -> t -> t
+(** Integer remainder. If [y] is not zero, the result of [Int16.rem x y]
+    satisfies the following property:
+    [x = Int16.add (Int16.mul (Int16.div x y) y) (Int16.rem x y)]. If [y = 0],
+    [Int16.rem x y] raises [Division_by_zero]. *)
+
+val unsigned_rem : t -> t -> t
+(** Same as {!rem}, except that arguments and result are interpreted as {e
+    unsigned} 16-bit integers. *)
+
+val succ : t -> t
+(** Successor. [Int16.succ x] is [Int16.add x Int16.one]. *)
+
+val pred : t -> t
+(** Predecessor. [Int16.pred x] is [Int16.sub x Int16.one]. *)
+
+val abs : t -> t
+(** [abs x] is the absolute value of [x]. On [min_int] this is [min_int] itself
+    and thus remains negative. *)
+
+val max_int : t
+(** The greatest representable 16-bit integer, 2{^15} - 1. *)
+
+val min_int : t
+(** The smallest representable 16-bit integer, -2{^15}. *)
+
+val logand : t -> t -> t
+(** Bitwise logical and. *)
+
+val logor : t -> t -> t
+(** Bitwise logical or. *)
+
+val logxor : t -> t -> t
+(** Bitwise logical exclusive or. *)
+
+val lognot : t -> t
+(** Bitwise logical negation. *)
+
+val shift_left : t -> int -> t
+(** [Int16.shift_left x y] shifts [x] to the left by [y] bits. The result is
+    unspecified if [y < 0] or [y >= 16]. *)
+
+val shift_right : t -> int -> t
+(** [Int16.shift_right x y] shifts [x] to the right by [y] bits. This is an
+    arithmetic shift: the sign bit of [x] is replicated and inserted in the
+    vacated bits. The result is unspecified if [y < 0] or [y >= 16]. *)
+
+val shift_right_logical : t -> int -> t
+(** [Int16.shift_right_logical x y] shifts [x] to the right by [y] bits. This is
+    a logical shift: zeroes are inserted in the vacated bits regardless of the
+    sign of [x]. The result is unspecified if [y < 0] or [y >= 16]. *)
+
+val of_int : int -> t
+(** Convert the given integer (type [int]) to a 16-bit integer (type [t]). The
+    argument is taken modulo 2{^16}. *)
+
+external to_int : t -> int = "%identity"
+(** Convert the given 16-bit integer (type [t]) to an integer (type [int]). *)
+
+val unsigned_to_int : t -> int option
+(** Same as {!to_int}, but interprets the argument as an {e unsigned} integer.
+    Returns [None] if the unsigned value of the argument cannot fit into an
+    [int]. *)
+
+val of_string : string -> t
+(** Convert the given string to a 16-bit integer. The string is read in decimal
+    (by default, or if the string begins with [0u]) or in hexadecimal, octal or
+    binary if the string begins with [0x], [0o] or [0b] respectively.
+
+    The [0u] prefix reads the input as an unsigned integer in the range
+    [[0, 2*Int16.max_int+1]]. If the input exceeds {!Int16.max_int} it is
+    converted to the signed integer [Int16.min_int + input - Int16.max_int - 1].
+
+    The [_] (underscore) character can appear anywhere in the string and is
+    ignored.
```

Comment:
```suggestion
    The [_] (underscore) character can appear anywhere after the prefix 
    in the string and is ignored.
```

Reply:
I checked it with `Int32` and `Int64` and the behaviour is the same. I'll send a correction PR for all the relevant modules after this one is merged.